geo-types/src/line_string.rs

```diff
@@ -2,9 +2,25 @@ use crate::{Coordinate, CoordinateType, Line, Point, Triangle};
 use std::iter::FromIterator;
 use std::ops::{Index, IndexMut};
 
-/// An ordered collection of two or more [`Coordinate`s](struct.Coordinate.html), representing a
+/// An ordered collection of two or more
+/// [`Coordinate`s](struct.Coordinate.html), representing a
 /// path between locations.
 ///
+/// A `LineString` is _closed_ if it is empty, or if the
+/// first and last coordinates are the same. The _boundary_
+/// of a `LineString` is empty if closed, and otherwise the
+/// end points. The interior is the (infinite) set of all
+/// points along the linestring _not including_ the
+/// boundary.
+///
+/// # Validity
+///
+/// A `LineString` is valid if it is either empty or
+/// contains 2 or more coordinates. Further, a closed
+/// `LineString` must not self intersect. Note that the
+/// validity is not enforced, and the operations and
+/// predicates are undefined on invalid linestrings.
+///
```

Comment:
Thanks for continuing to fleshing out the docs. :+1:

Reply:
+1